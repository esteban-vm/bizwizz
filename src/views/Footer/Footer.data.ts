import type { NavLinkProps } from '@/types'

const navLinks: NavLinkProps[] = [
  {
    id: '476b85cb-1e4d-4e03-9b66-d6ba167e09b2',
    className: 'nav_link footer',
    link: '#contact',
    text: 'Terms of use',
  },
  {
    id: '72012561-b194-4711-b2a3-f4591312ef94',
    className: 'nav_link footer',
    link: '#services',
    text: 'Privacy Policy',
  },
  {
    id: '8efd4a0e-7c40-476f-8c8a-c654e06a86e9',
    className: 'nav_link footer',
    link: '#portfolio',
    text: 'Cookie Policy',
  },
]

export default navLinks
