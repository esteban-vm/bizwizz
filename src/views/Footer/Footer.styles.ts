import clsx from 'clsx'

const styles = {
  wrapper: clsx('flex justify-between p-[3rem_0]'),
  copyright: clsx('text-darkgrey'),
  links: clsx('flex'),
  author: clsx('text-white'),
}

export default styles
