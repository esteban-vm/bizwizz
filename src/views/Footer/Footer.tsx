import type { FC } from '@/types'
import { NavLink } from '@/components'
import navLinks from './Footer.data'
import styles from './Footer.styles'

const Footer: FC = () => {
  return (
    <footer className={styles.wrapper}>
      <div className={styles.copyright}>
        <p>
          This website has been created by
          <a className={styles.author} href='#foo'>
            Esteban V.M.
          </a>
        </p>
      </div>
      <nav>
        <ul className={styles.links}>
          {navLinks.map((link) => (
            <NavLink key={link.id} {...link} />
          ))}
        </ul>
      </nav>
    </footer>
  )
}

export default Footer
