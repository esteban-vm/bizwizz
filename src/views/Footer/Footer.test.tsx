import { render, cleanup, within, screen } from '@/tests'
import Footer from './Footer'
import navLinks from './Footer.data'

describe('Footer:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Footer />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const footerElem = screen.getByRole('contentinfo')
    expect(footerElem).toBeInTheDocument()
    expect(within(footerElem).getByText(/this website has been created by/i)).toBeInTheDocument()
    expect(within(footerElem).getByRole('link', { name: /Esteban V\.M\./ })).toBeInTheDocument()
    expect(within(footerElem).getByRole('navigation')).toBeInTheDocument()
    expect(within(footerElem).getByRole('list')).toBeInTheDocument()
    expect(within(footerElem).getAllByRole('listitem')).toHaveLength(navLinks.length)
  })
})
