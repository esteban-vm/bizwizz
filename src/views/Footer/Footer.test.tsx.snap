// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer: > should render correctly 1`] = `
<DocumentFragment>
  <footer
    class="flex justify-between p-[3rem_0]"
  >
    <div
      class="text-darkgrey"
    >
      <p>
        This website has been created by
        <a
          class="text-white"
          href="#foo"
        >
          Esteban V.M.
        </a>
      </p>
    </div>
    <nav>
      <ul
        class="flex"
      >
        <li
          class="nav_link footer"
          id="476b85cb-1e4d-4e03-9b66-d6ba167e09b2"
        >
          <a
            href="#contact"
          >
            Terms of use
          </a>
        </li>
        <li
          class="nav_link footer"
          id="72012561-b194-4711-b2a3-f4591312ef94"
        >
          <a
            href="#services"
          >
            Privacy Policy
          </a>
        </li>
        <li
          class="nav_link footer"
          id="8efd4a0e-7c40-476f-8c8a-c654e06a86e9"
        >
          <a
            href="#portfolio"
          >
            Cookie Policy
          </a>
        </li>
      </ul>
    </nav>
  </footer>
</DocumentFragment>
`;
